src/mods/friendsList/index.js

```diff
@@ -16,82 +14,113 @@ function customFriendsList() {
+			const name = clickEvent.target.getAttribute('data-player-name');
+			player.whisperPlayer(name);
 		});
-		Array.from(document.querySelectorAll('.js-party-player')).forEach($button => {
-			$button.addEventListener('click', clickEvent => {
-				const name = clickEvent.target.getAttribute('data-player-name');
-				player.partyPlayer(name);
-			});
+	});
+	Array.from(document.querySelectorAll('.js-party-player')).forEach($button => {
+		$button.addEventListener('click', clickEvent => {
+			const name = clickEvent.target.getAttribute('data-player-name');
+			player.partyPlayer(name);
 		});
-		Array.from(document.querySelectorAll('.js-unfriend-player')).forEach($button => {
-			$button.addEventListener('click', clickEvent => {
-				const name = clickEvent.target.getAttribute('data-player-name');
-				player.unfriendPlayer(name);
+	});
+	Array.from(document.querySelectorAll('.js-unfriend-player')).forEach($button => {
+		$button.addEventListener('click', clickEvent => {
+			const name = clickEvent.target.getAttribute('data-player-name');
+			player.unfriendPlayer(name);
 
-				// Remove the blocked player from the list
-				Array.from(
-					document.querySelectorAll(`.js-friends-list [data-player-name="${name}"]`),
-				).forEach($element => {
-					$element.parentNode.removeChild($element);
-				});
-			});
-		});
-		Array.from(document.querySelectorAll('.js-friend-note')).forEach($element => {
-			$element.addEventListener('change', clickEvent => {
-				const name = clickEvent.target.getAttribute('data-player-name');
-				state.friendNotes[name] = clickEvent.target.value;
+			// Remove the blocked player from the list
+			Array.from(
+				document.querySelectorAll(`.js-friends-list [data-player-name="${name}"]`),
+			).forEach($element => {
+				$element.parentNode.removeChild($element);
 			});
 		});
-
-		// The close button for our custom UI
-		document.querySelector('.js-close-custom-friends-list').addEventListener('click', () => {
-			const $friendsListWindow = document.querySelector('.js-friends-list');
-			$friendsListWindow.parentNode.removeChild($friendsListWindow);
+	});
+	Array.from(document.querySelectorAll('.js-friend-note')).forEach($element => {
+		$element.addEventListener('change', clickEvent => {
+			const name = clickEvent.target.getAttribute('data-player-name');
+			state.friendNotes[name] = clickEvent.target.value;
 		});
 	});
+
+	// The close button for our custom UI
+	document
+		.querySelector('.js-close-custom-friends-list')
+		.addEventListener('click', hideFriendsList);
+}
+
+function hideFriendsList() {
+	const $friendsListWindow = document.querySelector('.js-friends-list');
+	$friendsListWindow.parentNode.removeChild($friendsListWindow);
 }
 
 export default {
 	name: 'Friends list',
 	description: 'Allows access to your friends list from the top right F icon',
-	run: customFriendsList,
+	run: ({ registerOnStateChange }) => {
+		customFriendsList();
+
+		registerOnStateChange((oldState, newState) => {
```

Comment:
Hm, so I contemplated this with implementing this hook, but am thinking more and more that it may be a really good thing to have:

What do you think about having a second argument to `registerOnStateChange`, that'd look like
`registerOnStateChange(callback, 'friendsList')`. Then only updates to the `friendsList` state key would trigger this callback.

Otherwise, we'll be running everything up to the `if (newlyFriendlyPlayers.length ...)` line every single state update, and we update state a decent amount of times now, and it may happen more often in the future.

(I'm definitely not asking you to implement this btw :P. Asking your opinion on me updating the code to handle this prior to us getting in this PR.)